examples/with-okta-multi-tenant-pkce-flow/api-server/okta.go

```diff
@@ -0,0 +1,144 @@
+	OktaDomain   string
+}
+
+var oktaVerifiers = sync.Map{}
+
+func newState() string {
+	const chars = "0123456789abcdefghijklmnopqrstuvwxyzABCDEFGHIJKLMNOPQRSTUVWXYZ"
+	bytes := make([]byte, 64)
+	if _, err := rand.Read(bytes); err != nil {
+		return ""
+	}
+
+	for i, b := range bytes {
+		bytes[i] = chars[b%byte(len(chars))]
+	}
+	return string(bytes)
+}
+
+var oktaProvider = tpmodels.TypeProvider{
+	ID:        "okta",
+	IsDefault: true,
+	Get: func(redirectURI, authCodeFromRequest *string, userContext supertokens.UserContext) tpmodels.TypeProviderGetResponse {
+		config, _ := (*userContext)["oktaConfig"].(OktaConfig)
+		// tenantId, _ := (*userContext)["tenantId"].(string)
+		oktaRedirectURI, _ := (*userContext)["redirectURI"].(string)
+
+		if authCodeFromRequest == nil {
+			empty := ""
+			authCodeFromRequest = &empty
+		}
+
+		state, _ := (*userContext)["state"].(string)
+		v, _ := verifier.CreateCodeVerifier()
+		verifierValue, _ := oktaVerifiers.LoadOrStore(state, v.Value) // save this to DB?
+		v.Value = verifierValue.(string)
+
+		return tpmodels.TypeProviderGetResponse{
+			AuthorisationRedirect: tpmodels.AuthorisationRedirect{
+				URL: fmt.Sprintf("https://%s/oauth2/default/v1/authorize", config.OktaDomain),
+				Params: map[string]interface{}{
+					"scope":                 "openid email",
+					"response_type":         "code",
+					"client_id":             config.ClientID,
+					"state":                 state,
+					"redirect_uri":          oktaRedirectURI,
+					"code_challenge":        v.CodeChallengeS256(),
+					"code_challenge_method": "S256",
+				},
+			},
+			AccessTokenAPI: tpmodels.AccessTokenAPI{
+				URL: fmt.Sprintf("https://%s/oauth2/default/v1/token", config.OktaDomain),
+				Params: map[string]string{
+					"client_id":     config.ClientID,
+					"grant_type":    "authorization_code",
+					"code":          *authCodeFromRequest,
+					"code_verifier": v.Value,
+					"redirect_uri":  oktaRedirectURI,
+				},
+			},
+			GetClientId: func(userContext supertokens.UserContext) string {
+				return config.ClientID
+			},
+			GetRedirectURI: func(userContext supertokens.UserContext) (string, error) {
+				return oktaRedirectURI, nil
+			},
+			GetProfileInfo: func(authCodeResponse interface{}, userContext supertokens.UserContext) (tpmodels.UserInfo, error) {
+				oauthTokens := authCodeResponse.(map[string]interface{})
+				authToken := oauthTokens["access_token"].(string)
+				userInfoAPIURL := fmt.Sprintf("https://%s/oauth2/default/v1/userinfo", config.OktaDomain)
+
+				userInfoRequest, err := http.NewRequest("GET", userInfoAPIURL, nil)
+				userInfoRequest.Header.Set("Authorization", fmt.Sprintf("Bearer %s", authToken))
+				if err != nil {
+					return tpmodels.UserInfo{}, err
+				}
+				userInfoResponse, err := doGetRequest(userInfoRequest)
+				if err != nil {
+					return tpmodels.UserInfo{}, err
+				}
+				userInfoMap := userInfoResponse.(map[string]interface{})
+				ID := userInfoMap["sub"].(string)
```

Comment:
Append the tenantId to this and also explain why thats being done -> it's cause the core doesn't support multiple user pools right now, and to emulate different user pools, we append the tenantId, so that the same okta user can login into different tenants.

Reply:
done